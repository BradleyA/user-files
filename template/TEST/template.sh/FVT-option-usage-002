#!/bin/bash
#	template/TEST/template.sh/FVT-option-usage-002  3.218.276  2019-07-28T11:41:08.765251-05:00 (CDT)  https://github.com/BradleyA/user-files.git  uadmin  six-rpi3b.cptx86.com 3.217  
#	   template.sh  add TEST cases 
### production standard 3.0 shellcheck
### production standard 5.1.160 Copyright
#       Copyright (c) 2019 Bradley Allen
#       MIT License is in the online DOCUMENTATION, DOCUMENTATION URL defined below.
BOLD=$(tput -Txterm bold)
NORMAL=$(tput -Txterm sgr0)
###

"${1}" -usage > "${0}".out
#
diff "${0}".expected "${0}".out >/dev/null 2>&1
RETURN_CODE=${?}
if [ ${RETURN_CODE} -eq 0 ] ; then
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - No difference with expected output - ${BOLD}PASS - PASS${NORMAL}"
elif [ ${RETURN_CODE} -eq 1 ] ; then
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Differences with expected output - ${BOLD}FAIL - FAIL${NORMAL}"
        diff -y "${0}".expected "${0}".out
else
        echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Test script ERROR - ${BOLD}FAIL - FAIL${NORMAL}"
fi
