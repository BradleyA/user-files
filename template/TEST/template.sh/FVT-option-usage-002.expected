FVT-option-usage-001.expected